azkaban-common/src/main/java/azkaban/flow/FlowProps.java

```diff
@@ -27,8 +27,20 @@
   private Props props = null;
 
   public FlowProps(final String parentSource, final String propSource) {
```

Comment:
Does this mean we have millions of FlowProps objects in the Heap simultaneously? Is that necessary?

Reply:
Yes, once a flow is brought in-memory, the FlowProps objects stay in until either the flow is removed or web server restarts. Changing that will require an overhaul.

Reply:
This pull request is super helpful and we should get it in.

I feel we should not just stop here. I think there have to be million(s) of unnecessary objects being created (and never deleted) for these duplicate string to turn into long-lived objects in the heap.